pkg/systemaccount/systemaccount.go

```diff
@@ -59,6 +63,18 @@ func (s *Manager) CreateSystemAccount(cluster *v3.Cluster) error {
 		RoleTemplateName: clusterOwnerRole,
 	})
 
+	// wait for initialRolePopulated for this user
+	for i := 0; i < 10; i++ {
+		user, err = s.users.Get(user.Name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if v3.UserConditionInitialRolesPopulated.IsTrue(user) {
```

Comment:
this annotation on the user does not indicate that the above CRTB has been applied. just indicates that the intial "bootstrapping" global role that allows a user to see himself has completed. There is an annotation on the above CRTB that you could check for instead. it would be added by norman that the create function of the `cluster-crtb-sync` life cycle had successfully excuted.

Note that I dont like the sleep in the loop, but still thinking on a better approach, just wanted to share this sooner rather than later.

Reply:
How about instead of a sleep loop, you create or get he crtb and check for the annotation. If it has the annotation, return true, else return false. Then, in the function that calls this one, if this function returned false, re-enqueue the cluster and exit

Reply:
:+1:

Reply:
actually we can just return error. This method is executed under `condition.DoUntilTrue`